test/suite/test_prepare27.py

```diff
@@ -0,0 +1,110 @@
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wttest
+from wtscenario import make_scenarios
+
+# test_prepare27.py
+# Test that a prepared update that has been aborted is not selected as the base value.
+class test_prepare27(wttest.WiredTigerTestCase):
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('column-fix', dict(key_format='r', value_format='8t')),
+        ('integer-row', dict(key_format='i', value_format='S')),
+        ('string-row', dict(key_format='S', value_format='S')),
+    ]
+    scenarios = make_scenarios(format_values)
+
+    def create_key(self, i):
+        return str(i) if self.key_format == 'S' else i
+
+    def create_value(self, i):
+        return str(i) if self.value_format == 'S' else i
+
+    def evict_cursor(self, uri, session_cfg, nrows):
+        s = self.conn.open_session()
+        s.begin_transaction(session_cfg)
+        evict_cursor = s.open_cursor(uri, None, "debug=(release_evict)")
+        for i in range(1, nrows + 1):
+            evict_cursor.set_key(self.create_key(i))
+            evict_cursor.search()
+            evict_cursor.reset()
+        s.rollback_transaction()
+        evict_cursor.close()
+
+    def test_prepare33(self):
+        uri = 'table:test_prepare27'
+        create_params = 'key_format={},value_format={}'.format(self.key_format, self.value_format)
+        self.session.create(uri, create_params)
+
+        # Apply a series of updates to a key.
+        num_updates = 5
+        key = self.create_key(1)
+
+        cursor = self.session.open_cursor(uri)
+        for ts in range(1, num_updates + 1):
+            with self.transaction(commit_timestamp = ts):
+                value = ts
+                cursor[key] = self.create_value(value)
+        
+        # Set the stable timestamp.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(2))
+    
+        # At this point we have 5 updates associated with a key in the update chain:
+        # 1 -> 2 -> 3 -> 4 -> 5
+
+        # By evicting, the latest update is moved to the DS and the rest to the HS.
+        self.evict_cursor(uri, None, 1)
+
+        # Perform a new update through a prepared transaction and evict the page.
+        num_updates = num_updates + 1
+        ts = num_updates
+
+        self.session.begin_transaction()
+        cursor[key] = self.create_value(ts)
+        self.session.prepare_transaction('prepare_timestamp=' + str(ts))
+
+        # Note that we need to configure the session to ignore prepared updates to be able to
+        # perform eviction. After eviction, we should have the prepared update on the DS and the
+        # rest in the HS.
+        self.evict_cursor(uri, "ignore_prepare=true", 1)
```

Comment:
This eviction is not required for the test to reproduce. It can be removed.

Reply:
By removing this eviction, I cannot reproduce it. However, by removing the first one line 85, I can reproduce the issue.
I think this makes sense, we need to have an eviction *at the very end* to have the last value on disk and the rest in the HS. What's important is to have the right content on the DS and HS before RTS and this can be done through eviction.
Let me know if you agree @kommiharibabu.

Reply:
I removed the unnecessary eviction in 2011d978725de281db2cacc85ba4d41de875b518. Let me know if you think that's wrong.